### DISCUSSION TITLE
New attribute in level spec to define tileset - Backwards Compatibility

### DISCUSSION THREAD

**Discussion by autofly17** [SuperTux/supertux#2436] [Ideas]
_(I am not a programmer, and this is my first-time using GitHub, so bear with me)_

As an idea to preserve backwards compatibility with older/unmaintained contribute levels, I propose adding a new attribute to the level spec that defines whether the level uses the old (0.6.3) or new (Nightly) tilesets.

My design proposal goes like the follows:
- Add a `newtiles` bool, which we will include and set to `#t` for already-converted levels.
- Levels that don't have `newtiles` will default to the old tileset.
- Attempting to edit an old tiles level will create and edit an alias with `newtiles` set to `#t` (With the appropriate warning message displayed, of course).

However, there are a few drawbacks to my proposal that I'd also like to mention:
- The installation would have to come with the old tiles included, which would be an added 7.36 MB
- `newtiles` is not a forwards-compatible name; if we decide to do another tileset overhaul (unlikely), the naming convention may get awkward.

**Comment by Semphriss** [comment]
It is already planned to remap the new tile graphics to the old format.

**Reply by Semphriss** [reply]
Update: #2462 is now open, though still WIP. I'll post more upates there.
